src/sonic_ax_impl/mibs/ietf/rfc1213.py

```diff
@@ -41,6 +44,53 @@ class DbTables(int, Enum):
     # ifOutQLen ::= { ifEntry 21 }
     SAI_PORT_STAT_IF_OUT_QLEN = 21
 
+class ArpUpdater(MIBUpdater):
+    def __init__(self):
+        super().__init__()
+        self.arp_dest_map = {}
+        self.arp_dest_list = []
+        # call our update method once to "seed" data before the "Agent" starts accepting requests.
+        self.update_data()
+
+    def update_data(self):
+        self.arp_dest_map = {}
+        self.arp_dest_list = []
+        for entry in python_arptable.get_arp_table():
+            dev = entry['Device']
+            mac = entry['HW address']
+            ip = entry['IP address']
+
+            if_index = mibs.get_index_from_str(dev)
+            if if_index is None: continue
+
+            mactuple = mac_decimals(mac)
+            machex = ''.join(chr(b) for b in mactuple)
+            # if MAC is all zero
+            #if not any(mac): continue
```

Comment:
The correct behavior of incomplete ARP entry (an IP without MAC) is not clear in the protocol. ref: [http://cric.grenoble.cnrs.fr/Administrateurs/Outils/MIBS/?oid=1.3.6.1.2.1.4.22.1.2](http://cric.grenoble.cnrs.fr/Administrateurs/Outils/MIBS/?oid=1.3.6.1.2.1.4.22.1.2)
Just keep the code here for further refinement, if needed.

---
In reply to: [108499363](https://github.com/Azure/sonic-snmpagent/pull/19#discussion_r108499363) [](ancestors = 108499363)